modules/ephemeris/src/main/scala/gem/horizons/HorizonsEphemerisUpdater.scala

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem
+
+package horizons
+
+import gem.dao.EphemerisDao
+import gem.enum.Site
+import gem.math.Ephemeris
+import gem.util.InstantMicros
+import gem.horizons.VelocityCompression._
+
+import cats._
+import cats.effect._
+import cats.effect.implicits._
+import cats.implicits._
+
+import doobie._, doobie.implicits._
+
+import java.time.{ Duration, Period }
+
+import fs2.Stream
+
+import scala.math.Ordering.Implicits._
+
+/** Utility for inserting / updating en ephemeris. */
+final case class HorizonsEphemerisUpdater(
+                   user: User[_],
+                   log:  Log[ConnectionIO],
+                   xa:   Transactor.Aux[IO, Unit]) {
+
+  import HorizonsEphemerisUpdater._
+
+  /** Constructs an action that when run will provide the current context
+    * information for an ephemeris.  This information can be used to determine
+    * whether an update is needed, to see when the last update was performed,
+    * and when the last update check happened.
+    */
+  def report(key:  EphemerisKey.Horizons, site: Site): IO[Context] =
+
+    for {
+
+      meta <- log.log(user, s"EphemerisDao.selectMeta($key, $site)") {
+                EphemerisDao.selectMeta(key, site)
+              }.transact(xa)
+
+      rnge <- log.log(user, s"EphemerisDao.selectTimes($key, $site)") {
+                EphemerisDao.selectTimes(key, site)
+              }.transact(xa)
+
+      soln <- log.log(user, s"HorizonsSolutionRefQuery($key).lookup") {
+                HorizonsSolutionRefQuery(key).lookup.liftIO[ConnectionIO]
+              }.transact(xa)
+
+    } yield Context(key, site, meta, rnge, soln)
```

Comment:
So, as this stands we have three transactions here and I'm not sure if that's what we want. It seems like at least the first two operations should be done at once. The third operation is only in `ConnectionIO` to placate the logger so it need not have a connection at all. We can leave it as-is for now but we should open an issue to fix it. I tried but ran into a cats bug I think.